packages/respect-core/src/modules/checks/checks-errors.ts

```diff
@@ -0,0 +1,29 @@
+import type { CHECKS } from './checks.js';
+
+export type ChecksErrorsType = keyof typeof CHECKS;
+
+export class ChecksErrors extends Error {
+  public readonly type: ChecksErrorsType;
+  public readonly originalError?: Error;
+
+  constructor(type: ChecksErrorsType, message: string, originalError?: Error) {
+    super(message);
+    this.name = 'ChecksErrors';
+    this.type = type;
+    this.originalError = originalError;
+  }
+}
+
+export class StatusCodeError extends ChecksErrors {
+  constructor(message: string, originalError?: Error) {
+    super('STATUS_CODE_CHECK', message, originalError);
+    this.name = 'StatusCodeError';
+  }
+}
+
```

Comment:
Do we use the originalError anywhere? Let's keep it simple.

```suggestion
export class ChecksErrors extends Error {}

export class StatusCodeError extends ChecksErrors {}

```

Reply:
You can also move this to the `./checks.ts` file itself.

Reply:
good, we can simplify.
Thx